packages/algob/src/internal/deployer_cfg.ts

```diff
@@ -0,0 +1,63 @@
+import { mkAccountIndex } from "../lib/account";
+import { AlgoOperator } from "../lib/algo-operator";
+import { loadASAFile } from "../lib/asa";
+import { loadCheckpointsRecursive } from "../lib/script-checkpoints";
+import type {
+  Accounts,
+  AlgobDeployer,
+  AlgobRuntimeEnv,
+  ASADefs,
+  CheckpointRepo
+} from "../types";
+import { DeployerDeployMode, DeployerRunMode } from "./deployer";
+import { txWriter, TxWriterImpl } from "./tx-log-writer";
+
+export interface DeployerConfig {
+  runtimeEnv: AlgobRuntimeEnv
+  cpData: CheckpointRepo
+  asaDefs: ASADefs
+  algoOp: AlgoOperator
+  accounts: Accounts
+  txWriter: txWriter
+}
+
+export function mkDeployer (
+  allowWrite: boolean,
+  deployerCfg: DeployerConfig
+): AlgobDeployer {
+  if (allowWrite) {
+    return new DeployerDeployMode(
+      deployerCfg.runtimeEnv,
+      deployerCfg.cpData,
+      deployerCfg.asaDefs,
+      deployerCfg.algoOp,
+      deployerCfg.accounts,
+      deployerCfg.txWriter);
+  }
+  return new DeployerRunMode(
+    deployerCfg.runtimeEnv,
+    deployerCfg.cpData,
+    deployerCfg.asaDefs,
+    deployerCfg.algoOp,
+    deployerCfg.accounts,
+    deployerCfg.txWriter);
+}
+
+// intialize deployer config obj
+export class MkDeployerConfig implements DeployerConfig {
```

Comment:
Last comment. 
This class is  a structure without methods. We don't need an interface. Let's remove `DeployerConfig` interface and rename `MkDeployerConfig` to `DeployerConfig`. Class names should be nouns or noun expression

Reply:
Makes sense. Updated